### PR TITLE
 #1_Aug_18_Api Created solution with an API project and BLL & DAL projects connected

### PR DESCRIPTION
**Pull request by REtangler**
-deleted a ton of unnecessary files from both BLL and DAL projects
-rewritten architecture a bit
-connected SQL server database from DAL project
-implemented API for Categories

	Files worth looking at:

-AltexFood_Delivery.API:
---CategoriesController.cs
---IServiceCollectionExtensions.cs

-AltexFood_Delivery.BLL:
---CategoriesService.cs

-AltexFood_Delivery.DAL:
---CategoryRepository.cs
---Repository.cs
---UnitOfWork.cs
---DataContext.cs
---DbFactory.cs

List of REST endpoints (for categories only):
-GET
-GET {id}
-POST
-PUT
-DELETE

### CODE REVIEW

**Comment by REtangler** [comment]
Updated build action to check the new solution instead of 3rd module project only

**Comment by REtangler** [comment]
It seems that i've unknowingly completed task 2 and if so, will it count as two tasks in this PR?
I can create second PR with more endpoints as i only have 5 in here.

File: DeliveryService_EF/Interfaces/ICategoryRepository.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AltexFood_Delivery.DAL.Models;
+
+namespace AltexFood_Delivery.DAL.Interfaces
+{
+    public interface ICategoryRepository : IRepository<Category>
+    {
+        Category NewCategory(string name, string description);
```
Comment (yevhenbilushko) [inline DeliveryService_EF/Interfaces/ICategoryRepository.cs:12]: maybe "Add" or "AddCategory"

Comment (yevhenbilushko) [inline DeliveryService_EF/Interfaces/ICategoryRepository.cs:12]: seems like better to use model as paarameter

File: DeliveryService_EF/Migrations/DataContextModelSnapshot.cs [hunk 2]
```diff
@@ -1,12 +1,12 @@
 ﻿// <auto-generated />
 using System;
-using DeliveryService_EF.Data;
+using AltexFood_Delivery.DAL.Data;
```
Comment (yevhenbilushko) [inline DeliveryService_EF/Migrations/DataContextModelSnapshot.cs:3]: we do not need migrations for now
remove it
or let it be if it is for filling default data

File: DeliveryService_EF/Program.cs [hunk 3]
```diff
@@ -1,22 +1,11 @@
 ﻿using System;
-using DeliveryService_EF.HwTasks;
-using DeliveryService_EF.HWtasks;
 
-namespace DeliveryService_EF
+namespace AltexFood_Delivery.DAL
 {
     public class Program
```
Comment (yevhenbilushko) [inline DeliveryService_EF/Program.cs:5]: againg try to use class lib not console app

File: AltexFood Delivery/AltexFood Delivery/Controllers/CategoriesController.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        private readonly CategoriesService _controller;
+        private readonly IUnitOfWork _unitOfWork;
+
+        public CategoriesController(CategoriesService controller, IUnitOfWork unitOfWork)
+        {
+            _controller = controller;
+            _unitOfWork = unitOfWork;
+        }
+
+        // GET categories
```
[the remark this answers is not in the document]
Comment (REtangler) [reply AltexFood Delivery/AltexFood Delivery/Controllers/CategoriesController.cs:22]: i've thought that it is a good practice to mark controller's methods according to their api call, isn't it?

File: AltexFood Delivery/AltexFood Delivery/Controllers/CategoriesController.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+namespace AltexFood_Delivery.Api.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CategoriesController : ControllerBase
+    {
+        private readonly CategoriesService _controller;
+        private readonly IUnitOfWork _unitOfWork;
+
+        public CategoriesController(CategoriesService controller, IUnitOfWork unitOfWork)
```
[the remark this answers is not in the document]
Comment (REtangler) [reply AltexFood Delivery/AltexFood Delivery/Controllers/CategoriesController.cs:16]: forgot to rename this variable after renaming CategoriesService class

File: DeliveryService_EF/Migrations/DataContextModelSnapshot.cs (hunk 2 above)
Comment (REtangler) [reply DeliveryService_EF/Migrations/DataContextModelSnapshot.cs:3]: this is from previous module, i use an existing db to test api
so i'd rather it stay

File: DeliveryService_EF/Interfaces/ICategoryRepository.cs (hunk 1 above)
Comment (REtangler) [reply DeliveryService_EF/Interfaces/ICategoryRepository.cs:12]: i've used model in 3rd PR for Product repo, so i'll leave product as is for this branch

Comment (REtangler) [reply DeliveryService_EF/Interfaces/ICategoryRepository.cs:12]: i've renamed Product repo method in 3rd PR, so i'll leave product as is for this branch.

File: AltexFood Delivery/AltexFood Delivery/Controllers/CategoriesController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (yevhenbilushko) [reply AltexFood Delivery/AltexFood Delivery/Controllers/CategoriesController.cs:22]: no

[the remark this answers is not in the document]
Comment (yevhenbilushko) [reply AltexFood Delivery/AltexFood Delivery/Controllers/CategoriesController.cs:22]: _cs - rename it 
add more sense
maybe _categoriesService 
up to u

**Review by yevhenbilushko — APPROVED** [review]

**Pull request merged**